core/templates/services/image-upload-helper.service.ts

```diff
@@ -40,7 +40,7 @@ export class ImageUploadHelperService {
       Math.random().toString(36).substr(2, 10);
   }
 
-  convertImageDataToImageFile(dataURI: string): Blob {
+  convertImageDataToImageFile(dataURI: string): Blob | null {
```

Comment:
Why are we allowing `null` here?

Reply:
For this function we're trying to validate that the input we're given is actually an image. We have the input string that holds the raw data that we try to parse into a true image, which is supposed to be a Blob now. If it can't be parsed into Blob, I believe we should return null to indicate the input is invalid.

I noticed that this function was called in core/templates/components/forms/custom-forms-directives/thumbnail-uploader.component.ts#L139

```
saveThumbnailImageData(imageURI: string, callback: () => void): void {
    this.resampledFile = null;
    this.resampledFile = (
      this.imageUploadHelperService.convertImageDataToImageFile(
        imageURI));
    this.encodedImageURI = imageURI;
    if (this.resampledFile === null) {
      this.alertsService.addWarning('Could not get resampled file.');
      return;
    }
    this.postImageToServer(this.resampledFile, callback);
  }
```
it checks if convertImageDataToImageFile() returned null and if it does, it throws a warning and returns. Alternatively instead of allowing null return I believe we can throw the warning in convertImageDataToImageFile() and return nothing and we won't have to check `if (this.resampledFile === null) ` in saveThumbnailImageData(). But I believe allowing null return is best, let me know if you have any suggestions?

Reply:
Okay, that is a great analysis, thank you! Can you please add a code comment explaining that we return `null` when the image cannot be converted